### ISSUE TITLE
Equalizer does not work with >1 item in each column

### ISSUE THREAD

**Issue by adrianwhite** [zurb/foundation-sites#9883]
I'm not really sure if this is a bug or a instead feature request and I'm actually trying do do something Equalizer cannot do.

#### How to reproduce this bug:

1. Use Basic example from http://foundation.zurb.com/sites/docs/equalizer.html
2. Add more than one item of example `.callout` markup in each column

#### What should happen:

Equalizer should still work, producing similar visual layout as http://foundation.zurb.com/sites/docs/equalizer.html#equalize-by-row as each item has `.data-equalizer-watch`

#### What happened instead:

If you **only have one item in the first column** the items in the second and third column get the height of the tallest item. In this example I'd expect the second and third items in the second and third columns to have a different height from the first items in each column as they have `.data-equalizer-watch`. I've tried adding `data-equalize-by-row="true"` but this did not help.

If you have **more than one item in the first column**, Equalizer fails completely.

#### Test case:

As you can see on http://codepen.io/adrianwhite/pen/mWXVGp

#### Use-case

I have multiple items of differing type/category (news, events, blog posts) I want to list in columns to maintain their association. As such, markup like http://foundation.zurb.com/sites/docs/equalizer.html#equalize-by-row is not suitable.

**Comment by gpspake** [comment]
Ok, it took me a minute but I see what you mean. 
If you are equalizing elements in a single row across multiple columns, adding more than one item to the _first_ column breaks equalizer for the whole row. However, if you add more items to the second or third columns, elements across all columns are equalized as expected. This is true even if you move `data-equalizer data-equalize-on="medium"` to a parent container outside of the row. To demonstrate this, go to the last example in the codepen and delete the second item in the first column and you'll see everything equalize.

So you can't equalize a row with more than one item in the first column?
Definitely looks like a bug.

**Comment by adrianwhite** [comment]
Cheers for taking a look @gpspake.

You are correct. Though there is one more thing that is probably more a feature request. I would have expected the elements with the same offset-top value to have the same height. 
http://foundation.zurb.com/sites/docs/equalizer.html#equalize-by-row

> Equalizer splits the data-equalizer-watch elements into groups by checking their vertical offsets, and grouping ones with the same offset into a "row".
> 
> Be aware on what you set data-equalizer-watch, if the top position is different, Equalizer will interpret that as a new "row" and equalize accordingly.

I do not think the current implementation can handle that as, as far as I can tell, it groups elements before applying any height. For the outcome I desire it would have to create the next group only after applying the height to the previous group.

**Comment by gpspake** [comment]
I thought about mentioning that but I didn't want to clutter up my response. 
This issue identifies an apparent bug but you also bring up a potentially valuable feature request. I would keep the two separate.

Your codepen examples all equalize to a single height (the bug is concerned with this scenario). If you want to equalize to multiple heights in the the same container, you have two options: row-equlaizer or nested-equalizer.


Row-equlaizer is for a totally different scenario where you are concerned with equalizing components across multiple rows rather than across columns - which isn't gonna help you considering your use case. It sounds like you want to be able to equalize all of the first elements across a group of columns, the second elements across a group of columns, etc.

This seems like a valid feature request because
1. It seems like a totally reasonable use case and
2. You can accomplish this now with some nested grid hackery but it requires a parent container for each (pseudo) row of items in your columns and gets ugly fast.
Here's a [working example](http://codepen.io/gpspake/pen/vxRyMV?editors=1010) using your codepen
This could be a viable solution if you have a small, predictable number of items in each column. You could either hard code the container markup or generate it based on a count of the items in the column with the most items; then just give each item the appropriate equalizer attribute value.

Without looking at the code I would bet that a lot of the problems associated with some sort of equalize-rows-across-columns feature have already been solved by the row-equalizer implementation as it's obviously firing up multiple watchers as the number of rows changes.

**Comment by adrianwhite** [comment]
I understand about not wanting to clutter your response.
I'll move the feature request to a new issue.
Is there anything more you need from me regarding the bug covered in this issue?

**Comment by adrianwhite** [comment]
I've been looking at this issue and I think I have found the culprit.
https://github.com/zurb/foundation-sites/blob/develop/js/foundation.equalizer.js#L164
When you have more than 1 item in the first column `_isStacked` will return true as the first two items are stacked. This makes `_reflow` return false so Equalizer does nothing.

**Comment by kball** [comment]
@adrianwhite would you like to submit a PR to fix this issue? :)

**Comment by adrianwhite** [comment]
Now I am reconsidering as to whether this is a bug.
Is this not what `data-equalize-on-stack="true"` is for? `_isStacked` will return true as the first two items are stacked so use `data-equalize-on-stack="true"` in combination with `data-equalize-on="medium"` if equalization on small viewports is not desired.
So far as I can tell that produces the desired result: http://codepen.io/adrianwhite/pen/XMwjbZ

Maybe if the documentation expanded upon  uses of `data-equalize-on-stack` instead?

**Comment by kball** [comment]
Well... if not a bug I think what has happened is you've gotten into a limitation of how Equalizer was being thought about... I think i twas just being though of as things side by side, and stacking would by default mean you were on a small screen... perhaps really the future here is the generalization being discussed in https://github.com/zurb/foundation-sites/issues/9889

If so, perhaps you can add a PR for better docs and we can close this in favor of the more generalized equalization being discussed in #9889

**Comment by adrianwhite** [comment]
@kball, apologies for being quiet for a while. I'm very busy at the mo.
I agree and will look into creating a PR for a documentation update. Should I leave this issue open until that has been created and accepted?

**Comment by kball** [comment]
Sounds good, thanks @adrianwhite

**Comment by JeremyEnglert** [comment]
Sounds like this wasn't a bug. @adrianwhite, feel free to submit a docs update if you think it's worthwhile (we'd love it!).
